documentation/modules/adding-source-provider.adoc

```diff
@@ -26,37 +26,174 @@ You can add {a-rhv} source provider by using the {ocp} web console.
 
 * {manager} CA certificate, unless it was replaced by a third-party certificate, in which case, specify the {manager} Apache CA certificate
 endif::[]
+ifdef::ova[]
+= Adding an Open Virtual Appliance (OVA) source provider
+
+You can add Open Virtual Appliance (OVA) files that were created by VMware vSphere as a source provider by using the {ocp} web console.
+
+include::snippet_ova_tech_preview.adoc[]
+endif::[]
+ifdef::ostack[]
+= Adding an {osp} source provider
+
+You can add an {osp} source provider by using the {ocp} web console.
+
+[NOTE]
+====
+Migration using {osp} source providers only supports VMs that use only Cinder volumes.
+====
+endif::[]
+ifdef::cnv[]
+= Adding a Red Hat {virt} source provider
+
+You can use a Red Hat {virt} provider as both a source provider and destination provider.
+
+Specifically, the host cluster that is automatically added to a {virt} provider can be used as both a source procider and a destination provider.
+
+You can migrate VMs from the cluster that {project-short} is deployed on to another cluster or from a remote cluster to the cluster that {project-short} is deployed on.
+
+endif::[]
+ifdef::cnv2[]
+= Adding {a-virt} destination provider
+
+You can use a Red Hat {virt} provider as both a source provider and destination provider.
+
+Specifically, the host cluster that is automatically added to a {virt} provider can be used as both a source procider and a destination provider.
+
+You can also add another {virt} destination provider to the {ocp} web console in addition to the default {virt} destination provider, which is the provider where you installed {project-short}.
+
+You can migrate VMs from the cluster that {project-short} is deployed on to another cluster or from a remote cluster to the cluster that {project-short} is deployed on.
+
+.Prerequisites
+
+* You must have {a-virt} link:https://access.redhat.com/documentation/en-us/openshift_container_platform/{ocp-version}/html/authentication_and_authorization/using-service-accounts[service account token] with `cluster-admin` privileges.
+endif::[]
 
 .Procedure
 
 . In the {ocp} web console, click *Migration* -> *Providers for virtualization*.
 . Click *Create Provider*.
 ifdef::vmware[]
-. Select *VMware* from the *Provider type* list.
+. Click *vSphere*.
 . Specify the following fields:
 
-* *Provider name*: Name to display in the list of providers
-* *vCenter host name or IP address*: vCenter host name or IP address - if a certificate for FQDN is specified, the value of this field needs to match the FQDN in the certificate
-* *vCenter user name*: vCenter user, for example, `user@vsphere.local`
-* *vCenter password*: vCenter user password
-* *VDDK init image*: VDDKInitImage path
+* *Provider resource name*: Name to display in the list of providers.
```

Comment:
@ahadas question, we say here "Name to display in the list of providers" this is true, this is the name we will display, but IMHO it's more important to say that this is the k8s object name, WDYT ?

Reply:
note: if we update the text here, we need to update in all the providers as well

Reply:
yeah, mentioning the k8s object is maybe too detailed - but it's not only about displaying the UI as you say
@RichardHoch how about replacing it with somethig like "The name of the source provider"?

Reply:
Good point! Replaced throughout with "Name of the source provider" In this case, I'm using a sentence fragment on purpose.